web/app/view/dialog/User.js

```diff
@@ -175,6 +175,31 @@ Ext.define('Traccar.view.dialog.User', {
                     }
                 }
             }]
+        }, {
+            xtype: 'fieldset',
+            title: Strings.twofa,
+            collapsible: true,
+            collapsed: true,
+            items: [{
+                xtype: 'checkboxfield',
```

Comment:
This checkbox is there when in the server configuration the totp preference is user instead of all. The 'user' settings means we can turn totp on per user while 'all' means it is on for every user it cannot be turned off.
If we only want to have the 'user' option than this checkbox can go.

Reply:
Sorry, but I don't really understand the explanation here. If server configured as "all", it means totp is forced, so we don't need checkbox. If server configured as "user", then each user can set key or leave it empty to disable totp.

Reply:
When totp is enabled we have two scenarios in which it is active:
1st: Always active for every user. The checkbox is in this case indeed not required.
2nd: Can be turned on or off by the an admin per user. In this case the checkbox only needs to be active when an admin is logged in. 

The totp key is generated server side. The user should be able to clear it's own key so a new one is generated server side. An admin should also be able to clear a key so server side a new one will be generated for the user.

Reply:
OK, looks like I completely misunderstood it. I don't think we need admin to be able to set it per user. We want user to be able to enable 2FA for their account if they want to.

Reply:
What about this point?

Reply:
There are now tree options for 2FA/TOTP:

1: TOTP disabled. This is the default for backwards compatibility
2: TOTP enabled but not enforced. Users can enable using TOTP as 2FA by setting the checkbox in their account. The admin can also change this.
3: TOTP enabled and enforced. The checkbox will be hidden.